src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -160,10 +160,10 @@ protected final UserFilterType filterParam(final RestRequest request) {
     ValidationResult<String> withAuthTokenPath(final RestRequest request) throws IOException {
         return endpointValidator.withRequiredEntityName(nameParam(request)).map(username -> {
             // Handle auth token fetching
-            if (!(request.uri().contains("/internalusers/" + username + "/authtoken") && request.uri().endsWith("/authtoken"))) {
-                return ValidationResult.error(RestStatus.NOT_IMPLEMENTED, methodNotImplementedMessage(request.method()));
+            if (request.uri().contains("/internalusers/" + username + "/authtoken") && request.uri().endsWith("/authtoken")) {
```

Comment:
Is the second condition unnecessary, since the first one is already checking it?

Reply:
The original change in [3123](https://github.com/opensearch-project/security/pull/3123/files#diff-b097e68c9e3c8957ce28da0ad7e7fd93bf95c747eba154c9b9269edc2df55713L237) had a bug with the logic, and hence the inversion makes sense here. I am mainly talking about the necessity of the second condition, since the first one is already checking that it includes `/authtoken`. Maybe there is some edge case of not ending in `/authtoken`?